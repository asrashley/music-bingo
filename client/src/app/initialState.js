import { initialState as cards } from '../cards/cardsSlice';
import { initialState as games } from '../games/gamesSlice';
import { initialState as tickets } from '../tickets/ticketsSlice';
import { initialState as user } from '../user/userSlice';

export const initialState = Object.freeze({
    cards,
    games,
    tickets,
    user,
});
