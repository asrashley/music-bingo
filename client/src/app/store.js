import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import { connectRouter } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { reducer as formReducer } from 'redux-form';

import cardsReducer from '../cards/cardsSlice';
import gamesReducer from '../games/gamesSlice';
import ticketsReducer from '../tickets/ticketsSlice';
import userReducer from '../user/userSlice';
import adminReducer from '../admin/adminSlice';
import usersMiddleware from '../user/userMiddleware';

export const history = createBrowserHistory();

export const store = configureStore({
  middleware: [...getDefaultMiddleware(), usersMiddleware, logger],
  reducer: {
    admin: adminReducer,
    cards: cardsReducer,
    form: formReducer,
    games: gamesReducer,
    tickets: ticketsReducer,
    router: connectRouter(history),
    user: userReducer,
  },
});
