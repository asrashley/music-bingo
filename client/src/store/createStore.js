import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import { connectRouter } from 'connected-react-router';

import adminReducer from '../admin/adminSlice';
import directoriesReducer from '../directories/directoriesSlice';
import gamesReducer from '../games/gamesSlice';
import messagesReducer from '../messages/messagesSlice';
import settingsReducer from '../settings/settingsSlice';
import systemReducer from '../system/systemSlice';
import ticketsReducer from '../tickets/ticketsSlice';
import userReducer from '../user/userSlice';
import usersMiddleware from '../user/userMiddleware';
import messagesMiddleware from '../messages/messagesMiddleware';

import { history } from './history';

export function createStore(preloadedState, logging) {
  const middleware = [...getDefaultMiddleware(), usersMiddleware, messagesMiddleware];
  if (logging === undefined) {
    logging = (process.env.NODE_ENV === 'development');
  }
  if (logging) {
    middleware.push(logger);
  }
  return configureStore({
    middleware,
    reducer: {
      admin: adminReducer,
      directories: directoriesReducer,
      games: gamesReducer,
      messages: messagesReducer,
      settings: settingsReducer,
      system: systemReducer,
      tickets: ticketsReducer,
      router: connectRouter(history),
      user: userReducer,
    },
    preloadedState
  });
}
