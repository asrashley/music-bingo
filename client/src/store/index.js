export { createStore } from './createStore';
export { initialState } from './initialState';
export { history } from './history';
