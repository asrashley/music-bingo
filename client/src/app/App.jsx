import React from 'react';
import { ConnectedRouter } from 'connected-react-router';
import { Switch, Route } from 'react-router-dom';

import { NavPanel } from '../components/NavPanel';
import routes from '../routes';
import { IndexPage } from './IndexPage';
import { ListGamesPage, PastGamesPage, TrackListingPage } from '../games/components';
import { ChooseTicketsPage, PlayGamePage, ViewTicketPage } from '../tickets/components';
import {
  LoginPage, LogoutPage, PasswordResetPage, PasswordResetConfirmPage,
  LoginRequired,
} from '../user/components';
import { UsersListPage } from '../admin/components';
import { RegisterPage } from '../user/components/RegisterPage';
import { MessagePanel } from '../messages/components';
import { history } from './store';

import '../styles/main.scss';

const routeComponents = [
  { path: routes.login, component: LoginPage, exact:true },
  { path: routes.logout, component: LogoutPage, exact: true },
  { path: routes.register, component: RegisterPage, exact: true },
  { path: routes.passwordResetConfirm, component: PasswordResetConfirmPage, exact: true },
  { path: routes.passwordReset, component: PasswordResetPage, exact: true },
  { path: routes.listUsers, component: UsersListPage, exact: true, protected: true },
  { path: routes.listGames, component: ListGamesPage, exact: false },
  { path: routes.chooseTickets, component: ChooseTicketsPage, exact: true, protected: true },
  { path: routes.play, component: PlayGamePage, exact: true, protected: true },
  { path: routes.viewTicket, component: ViewTicketPage, exact: true, protected: true },
  { path: routes.pastGames, component: PastGamesPage, exact: true, protected: true },
  { path: routes.trackListing, component: TrackListingPage, exact: true, protected: true },
  { path: routes.index, component: IndexPage, exact: false },
];

function App() {
  return (
    <ConnectedRouter history={history}>
      <Switch>
        {routeComponents.map((route, index) => (
          <Route exact={route.exact} path={route.path} key={index} component={NavPanel} />
        ))}
      </Switch>
        <div className="container">
          <MessagePanel />
      <Switch>
          {routeComponents.map((route, index) => (
            <Route exact={route.exact} path={route.path} key={index} component={route.component} />
          ))}
      </Switch>
      <Switch>
        {routeComponents.filter(route => route.protected===true).map((route, index) => (
            <Route exact={route.exact} path={route.path} key={index} component={LoginRequired} />
        ))}
      </Switch>
        </div>
    </ConnectedRouter>
  );
}

export default App;
