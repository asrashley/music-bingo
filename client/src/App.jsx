import React from 'react';
import PropTypes from 'prop-types';
import { ConnectedRouter } from 'connected-react-router';
import { Redirect, Switch, Route } from 'react-router-dom';
import { reverse } from 'named-urls';
import { connect } from 'react-redux';

import { NavPanel } from './components/NavPanel';
import routes from './routes';
import { IndexPage } from './games/components';
import { ChooseTicketsPage, PlayGamePage, ManageGamePage } from './tickets/components';
import { LoginPage, LogoutPage } from './user/components';
import { RegisterPage } from './user/components/RegisterPage';
import { initialState } from './app/initialState';
import { history } from './app/store';

import './styles/main.scss';

function ProtectedRouteFn({ children, ...rest }) {
  return (
    <Route
      {...rest}
      render={(props) => {
        console.log('********** protected route');
        const { location } = props;
        console.dir(props);
        if (false) {
          return children;
        }
        return (
          <Redirect
            to={{
              pathname: reverse('login'),
              state: { from: location }
            }}
          />
        );

      }}
    />
  );
}

const mapStateToProps = (state) => {
  state = state || initialState;
  const { user } = state;
  return {
    user,
  };
};

const ProtectedRoute = connect(mapStateToProps)(ProtectedRouteFn);


function App() {
  return (
    <ConnectedRouter history={history}>
      <NavPanel />
      <Switch>
        <Route exact path={routes.login} component={LoginPage} />
        <Route exact path={routes.logout} component={LogoutPage} />
        <Route exact path={routes.register} component={RegisterPage} />
        <Route exact path={routes.game} component={ChooseTicketsPage} />
        <Route exact path={routes.manage} component={ManageGamePage} />
        <Route exact path={routes.play} component={PlayGamePage} />
        <Route path={routes.index} component={IndexPage} />
      </Switch>
    </ConnectedRouter>
  );
}

export default App;
