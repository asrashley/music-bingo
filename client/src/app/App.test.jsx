import React from 'react';
import PropTypes from 'prop-types';
import { render, queries } from '@testing-library/react';
import { it } from 'vitest';
import { Provider } from 'react-redux';

import { installFetchMocks, fetchMock } from '../../tests';
import { createStore } from '../store/createStore';
import { initialState } from '../store/initialState';
import App from './App';


function renderWithProviders(ui) {
  const store = createStore(initialState);
  function Wrapper({ children }) {
    return (<Provider store={store}>
      {children}
    </Provider>);
  }
  Wrapper.propTypes = {
    children: PropTypes.node.isRequired,
  };
  return {
    store, ...render(ui, {
      wrapper: Wrapper,
      queries,
    })
  };
}

describe('App', () => {
  beforeEach(() => {
    installFetchMocks(fetchMock, { loggedIn: true });
  });

  afterEach(() => {
    fetchMock.mockReset();
    vi.clearAllTimers();
    vi.useRealTimers();
  });

  it('App component matches snapshot', async () => {
    vi.useFakeTimers('modern');
    vi.setSystemTime(new Date('29 Dec 2023 19:12:00 GMT').getTime());
    const { asFragment, findByText } = renderWithProviders(<App />);
    await findByText('Privacy Policy');
    await findByText('admin');
    expect(asFragment()).toMatchSnapshot();
  });
})
