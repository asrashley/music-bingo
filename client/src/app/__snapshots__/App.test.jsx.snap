// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > App component matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="nav-bar"
  >
    <nav
      class="navbar navbar-expand navbar-light users creators hosts admin"
      id="nav-menu"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <img
          alt=""
          class="d-inline-block align-top"
          height="30"
          src="undefined/img/Icon.png"
          width="30"
        />
        Musical Bingo
      </a>
      <button
        aria-controls="nav-menu"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-target="#nav-menu"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          class="nav nav-pills"
        >
          <li
            class="nav-item active"
          >
            <a
              class="nav-link active"
              href="/"
            >
              Home 
              <span
                class="sr-only"
              >
                (current)
              </span>
            </a>
          </li>
          <li
            class="nav-item "
          >
            <a
              class="nav-link "
              href="/game"
            >
              Now Playing
            </a>
          </li>
          <li
            class="nav-item dropdown "
          >
            <button
              aria-expanded="false"
              class="nav-link dropdown-toggle  "
            >
              Previous Games
            </button>
            <ul
              class="dropdown-menu  "
            >
              <li>
                <a
                  class="dropdown-item"
                  href="/history"
                >
                  Popularity
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/history/themes"
                >
                  Theme's Last Use
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/history/calendar"
                >
                  Calendar
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/history/games"
                >
                  All Previous Games
                </a>
              </li>
            </ul>
          </li>
          <li
            class="nav-item "
          >
            <a
              class="nav-link "
              href="/clips"
            >
              Clips
            </a>
          </li>
          <li
            class="nav-item dropdown "
          >
            <button
              aria-expanded="false"
              class="nav-link dropdown-toggle  "
            >
              admin
            </button>
            <ul
              class="dropdown-menu  "
            >
              <li>
                <a
                  class="dropdown-item"
                  href="/user/users"
                >
                  Modify Users
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/user/settings"
                >
                  Modify Settings
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/user/guests"
                >
                  Guest links
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/user"
                >
                  User Profile
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/user/modify"
                >
                  Change password or email address
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/user/logout"
                >
                  Log out
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </nav>
    <nav
      aria-label="breadcrumb"
      class="navbar navbar-light breadcrumbs"
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item active"
        >
          <span>
            Home
          </span>
        </li>
      </ol>
    </nav>
    <div
      class="version-info"
    >
      <p
        class="version"
      >
         v0.2.5
      </p>
      <p
        class="hash"
      >
        914af0c
      </p>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      id="message-panel"
    />
    <div
      class="index-page"
    >
      <div
        class="logo"
      />
      <div
        class="welcome"
      >
        <h2
          class="strapline"
        >
          Like normal Bingo, but with music!
        </h2>
        <p
          class="description"
        >
          Musical Bingo is a variation on the normal game of bingo, where the numbers are replaced with songs that the players must listen out for.
        </p>
        <p
          class="description"
        >
          If you know your Bruce Springsteen from your Beyonce, your Whitney Houston from your Wu-Tang Clan, why not join use for a game or two?
        </p>
        <div
          class="description"
        >
          There are no upcoming Bingo games, but in the meantime you could browse the 
          <a
            href="/history"
          >
             previous Bingo games
          </a>
          .
        </div>
      </div>
      <div
        class="number footer"
      >
        Game 2023-12-29 / Ticket 20
      </div>
    </div>
  </div>
  <footer>
    <p
      class="footer"
    >
      <span
        class="copyright"
      >
        (c) 2020 Alex Ashley
      </span>
      <span
        class="github-link"
      >
        <a
          href="https://github.com/asrashley/music-bingo"
        >
          github.com/asrashley/music-bingo
        </a>
      </span>
      <span
        class="privacy-policy"
      >
        <a
          href="/privacy"
        >
          Privacy Policy
        </a>
      </span>
    </p>
  </footer>
</DocumentFragment>
`;
